### ISSUE TITLE
run-tests.js is broken for cmd.exe

### ISSUE THREAD

**Issue by nfischer** [shelljs/shelljs#294]
I was trying out `run-tests.js` on my Windows machine, and ran into *lots* of problems. Since we're trying to get good support for Windows, I figured I'd make this a formal issue so that others can chime in and confirm the behavior I'm seeing.

The end-goal of this issue is to open up `cmd.exe` on Windows, `cd` to the shelljs directory, type `npm test`, and see that all the tests pass. To me this seems like a worthy goal.

With that said, if someone else sees behavior different from what I'm seeing, please verify that you're running everything through `cmd.exe`, like I am. I already know this can run on some other shells (Bash, for instance), but I'd like to see this work on this particular shell.

Initial problems I've seen:

 - must use `node path/to/jshint` in the `run-tests` script, instead of relying on the shebang
 - must expand wildcards ourselves, not relying on the shell, since [cmd relies on the application to expand them](https://superuser.com/questions/460598/is-there-any-way-to-get-the-windows-cmd-shell-to-expand-wildcard-paths), and jshint doesn't appear to be doing this
 - `exec()` will start in the current working directory if run through `cmd.exe`. If we pass the `cwd` argument to `child_process.exec()`, then this will fix the issue only for admin mode (not regular user mode). I think this is only possible to fix for admin mode. This means that all our exec calls will need `pwd()` prefixed to the paths.

I'll add more to the list in comments as I find the issues.

**Comment by nfischer** [comment]
I'll open a PR later that at least gets jshint working properly (and should work on both Windows and Linux).

**Comment by mtscout6** [comment]
Have you considered getting a CI setup with [AppVeyor](https://www.appveyor.com/)? It's a free service like travis that runs on Windows machines. It also supports GitHub integration like Travis for Pull Requests so you know they work with both Windows and Linux based systems.

**Comment by nfischer** [comment]
Yes! We're working on setting up AppVeyor (although this issue sounds like it needs to be fixed before AppVeyor can be useful). If you have any experience writing an AppVeyor configure file, help would be appreciated :+1:

**Comment by mtscout6** [comment]
I did one for [react-bootstrap](https://github.com/react-bootstrap/react-bootstrap) which is:

https://github.com/react-bootstrap/react-bootstrap/blob/master/appveyor.yml

```

version: '{build}'

# Install scripts. (runs after repo cloning)
install:
  # Get the latest stable version of io.js
  - ps: Install-Product node ''
  - npm -g install npm@latest
  - set PATH=%APPDATA%\npm;%PATH%
  - node --version
  - npm --version
  - npm install

# No need for MSBuild on this project
build: off

test_script:
  - npm test
```

**Comment by nfischer** [comment]
Awesome! That looks great, thank you! @ariporad what do you think?

**Comment by nfischer** [comment]
@mtscout6 Would you be able to take a look at #298? I'm using the same yml, and the project is configured to only run AppVeyor on branches with the yml file. Unfortunately, I'm not seeing AppVeyor integration on the PR, although the page for it is https://ci.appveyor.com/project/nfischer/shelljs. Could it perhaps be ignoring the yml file? Thanks in advance!

**Comment by mtscout6** [comment]
I'll have to look at it tomorrow when I'm back in front of a computer.

**Comment by nfischer** [comment]
Thanks! It seems like it's only running on the master branch, not the PR, which might explain it

**Comment by mtscout6** [comment]
At first glance I think you're missing this:

![screen shot 2016-01-20 at 10 05 10 am](https://cloud.githubusercontent.com/assets/736870/12456506/6bcbeafc-bf5d-11e5-91db-8b8806f1910d.png)

Here's a screenshot of React-Bootstrap's configuration as well:

![screen shot 2016-01-20 at 10 09 50 am](https://cloud.githubusercontent.com/assets/736870/12456640/020e7624-bf5e-11e5-8bab-1e3014f83886.png)

It's odd some things can be controlled through the yaml file, but not everything. It's in typical Microsoft fassion.

**Comment by mtscout6** [comment]
If you're up for granting me access to that project in AppVeyor I can poke around and see if anything stands out to me.

**Comment by nfischer** [comment]
Yup, I was missing that. I double checked that our settings match yours and will try another commit later today to test. Thanks!

**Comment by mtscout6** [comment]
Another issue with why AppVeyor is not getting triggered is because you need to setup a webhook in Github.

Here's more of React-Bootstrap's configuration:

![screen shot 2016-01-20 at 5 13 46 pm](https://cloud.githubusercontent.com/assets/736870/12467530/4766c1dc-bf99-11e5-9e3f-8d45c99cc6c9.png)

**Comment by mtscout6** [comment]
There's are the specific details of that webhook:

![screen shot 2016-01-20 at 5 16 11 pm](https://cloud.githubusercontent.com/assets/736870/12467563/8d65fac2-bf99-11e5-99d7-83a52eb6c0a7.png)

**Comment by nfischer** [comment]
Ok, great! Unfortunately, I don't have access to the settings for this repo, so I won't be able to get that modified just yet. @ariporad @arturadib would one of you be able to grant access or set this up? I can provide the webhook.

**Comment by ariporad** [comment]
@nfischer: Nope, waiting on @arturadib to give both of us admin access to the shelljs org. Then we can.
